### PR TITLE
Docs: Add gc.enabled property

### PR DESCRIPTION
**Pull request by nk1506**

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ Iceberg tables support table properties to configure table behavior, like the de
 | history.expire.max-snapshot-age-ms | 432000000 (5 days) | Default max age of snapshots to keep on the table and all of its branches while expiring snapshots |
 | history.expire.min-snapshots-to-keep | 1                | Default min number of snapshots to keep on the table and all of its branches while expiring snapshots |
 | history.expire.max-ref-age-ms      | `Long.MAX_VALUE` (forever) | For snapshot references except the `main` branch, default max age of snapshot references to keep while expiring snapshots. The `main` branch never expires. |
+| gc.enabled                         | true             | Property to disable garbage collection operations such as expiring snapshots or remove orphan files |
```
Comment (ajantha-bhat) [inline docs/configuration.md:114]: should be "expire" snapshots or "removing" orphan files.

**Comment by ajantha-bhat** [comment]
PR title can be "gc.enabled" instead of "GC.ENABLED" as properties are case sensitive.

**Review by ajantha-bhat — APPROVED** [review]
LGTM. 

But not sure whether it was intentionally not documented as it is unlikely that the user wants to disable it.

**Comment by nk1506** [comment]
Closing this as this might nor required on doc.

**Pull request closed without merging**